pfsimulator/parflow_lib/problem_saturation.c

```diff
@@ -200,15 +217,15 @@ void     Saturation(
           {
             GrGeomInLoop(i, j, k, gr_solid, r, ix, iy, iz, nx, ny, nz,
             {
-              int ips = SubvectorEltIndex(ps_sub, i, j, k);
+              ips = SubvectorEltIndex(ps_sub, i, j, k);
```

Comment:
These do not compile with CUDA; the local variables inside loops must be locally defined, ie,

`int ips = SubvectorEltIndex(ps_sub, i, j, k);`